core/core_modules/disable.js

```diff
@@ -1,24 +1,107 @@
-				'What is best is you comply. Compliance will be rewarded. Are you ready to comply ' +
-				event.sender_name + '?', event.thread_id);
-		}
-		disabled = !disabled;
-	}
-	return false;
-};
+    if (event.arguments[0] === api.commandPrefix + commands[0] || threads[event.thread_id].possibleSpam) {
 
-exports.help = function(commandPrefix) {
-	return [[commandPrefix + 'disable','Disables the platform', 'Stops the platform from responding to messages it receives.\nTo renable send the disable command again']];
-};
+        // Command /disable /counter <value>
+        if (event.arguments_body.startsWith(api.commandPrefix + commands[1])) {
+            threads[event.thread_id].counterLimit = parseInt(event.arguments_body.substring(
+                (api.commandPrefix + commands[1]).length, event.arguments_body.length));
+                if (isNaN(threads[event.thread_id].counterLimit)) {
+                    threads[event.thread_id].counterLimit = 3;
+                    api.sendMessage(messages[4] + ' ' + event.sender_name, event.thread_id);
+                } else {
+                    api.sendMessage(messages[5] + ' ' + event.sender_name, event.thread_id);
+                }
+                return false;
+
+                // Command /disable /timer <seconds>
+            } else if (event.arguments_body.startsWith(api.commandPrefix + commands[2])) {
+                var seconds = parseFloat(event.arguments_body.substring(
+                    (api.commandPrefix + commands[2]).length, event.arguments_body.length));
+                    if (isNaN(seconds)) {
+                        api.sendMessage(messages[4] + ' ' + event.sender_name, event.thread_id);
+                    } else {
+                        setTimeout(function(){
+                            threads[event.thread_id].isThreadDisabled = !threads[event.thread_id].isThreadDisabled;
+                        }, seconds * 1000); // Converting seconds to milliseconds
+                        api.sendMessage(messages[5] + ' ' + event.sender_name, event.thread_id);
+                    }
+                    return false;
+
+                    // Command /disable /default
+                } else if (event.arguments_body === api.commandPrefix + commands[4]) {
+                    threads[event.thread_id].counterLimit = 3;
+                    return false;
+
+                    // Main Branch - only executed if no commands were matched
+                } else if (threads[event.thread_id].isThreadDisabled	) {
+                    api.sendMessage(messages[msgIndexEnable] + ' ' + event.sender_name, event.thread_id);
+                }
+                else {
+                    api.sendMessage(messages[msgIndexDisable], event.thread_id);
+                }
+                threads[event.thread_id].isThreadDisabled = !threads[event.thread_id].isThreadDisabled;
+                threads[event.thread_id].possibleSpam = false;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 8 space characters but found 16.](https://www.codacy.com/app/theprojectshop/Kassy/file/2556439885/issues/source?bid=3282987&fileBranchId=3282995#l99)